docs/contracts/using-for.rst

```diff
@@ -38,6 +48,13 @@ then the type (``uint``) has to be implicitly convertible to the
 first parameter of each of these functions. This check is
 performed even if none of these functions are called.
 
+If you define an operator for a user-defined type (``using {f as +} for T``), then
+the type (``T``), types of function parameters and the type of the function return value
+have to be the same. The type (``T``) does not include data location.
+But, data location of the function parameters and function return value must be
+the same. One exception from this is the result type of comparison and boolean operators
```

Comment:
This suggests that there can be data locations involved, even though we restricted to value-types.

Reply:
I'd generally try not to phrase this as "usually it's all the same type, but there are exceptions", but that each operator requires types of a specific generic signature and then list those signatures for all supported operators...

Reply:
Ah, good point. I missed that when extracting support for structs.

> I'd generally try not to phrase this as "usually it's all the same type, but there are exceptions" 

Yeah, I'll rephrase it.

BTW, These docs are what was originally in the PR and I left it mostly untouched but I'm not very satisfied with it. I'm planning to rewrite the whole section on `using for` once I'm done with suffixes. I wanted to leave it for later because I think the whole section could be clearer, not just the bits that were added for operators.

Reply:
Done. I added a table listing possible signatures. It's a bit verbose and repetitive so not sure how well it will when we add more operators to it. Still I kinda like it this way. Please take a look and tell me what you think.

Reply:
Eventually, the operators will actually have generic in-language definitions, then we can just link the standard library here instead. The verbose part is partly lack of https://github.com/ethereum/solidity/issues/1349 and lack of fixed syntax conventions for type variables :-). But lacking those, the current list is good!

Reply:
> partly lack of #1349

Oh, I wasn't even aware of that issue. Should I change to a notation that resembles this?

Reply:
No, I just meant in the future we can make it mire concise, but for now let's stick to actually valid syntax :-)